app/models/host/hostmix.rb

```diff
@@ -7,5 +7,10 @@ def has_many_hosts(options = {})
     def belongs_to_host(options = {})
       belongs_to :host, {:class_name => "Host::Managed", :foreign_key => :host_id}.merge(options)
     end
+
+    def hosts_count(id)
```

Comment:
We have a helper for the UI that handles the hosts counts for various pages and takes into consideration authorizations as well (https://github.com/theforeman/foreman/blob/develop/app/helpers/application_helper.rb#L493), and also caches the result to prevent the N+1 issue @ohadlevy mentioned - I suggest considering extracting it from the helper and reusing it for this.